Field alias AttributeError: 'AliasChoices' object has no attribute 'lower' - CliSettingsSource

### Description

Field aliases fail when subclassing a model with BaseSettings and using AliasChoices.

<details>

<summary>Pytest output</summary>

```shell
/usr/local/lib/python3.11/dist-packages/pydantic_settings/main.py:141: in __init__
    **__pydantic_self__._settings_build_values(
/usr/local/lib/python3.11/dist-packages/pydantic_settings/main.py:260: in _settings_build_values
    CliSettingsSource(
/usr/local/lib/python3.11/dist-packages/pydantic_settings/sources.py:902: in __init__
    self._connect_root_parser(
/usr/local/lib/python3.11/dist-packages/pydantic_settings/sources.py:1236: in _connect_root_parser
    self._add_parser_args(
/usr/local/lib/python3.11/dist-packages/pydantic_settings/sources.py:1255: in _add_parser_args
    for field_name, resolved_name, field_info in self._sort_arg_fields(model):
/usr/local/lib/python3.11/dist-packages/pydantic_settings/sources.py:1154: in _sort_arg_fields
    resolved_name = resolved_name.lower() if not self.case_sensitive else resolved_name
E   AttributeError: 'AliasChoices' object has no attribute 'lower'\

self = EnvSettingsSource(env_nested_delimiter='.', env_prefix_len=0), model = <class 'test_pydantic-settings.test_settings.<locals>.Check'>
    def _sort_arg_fields(self, model: type[BaseModel]) -> list[tuple[str, str, FieldInfo]]:
        positional_args, subcommand_args, optional_args = [], [], []
        fields = model.__pydantic_fields__ if is_pydantic_dataclass(model) else model.model_fields
        for field_name, field_info in fields.items():
            resolved_name = field_name if field_info.alias is None else field_info.alias
>           resolved_name = resolved_name.lower() if not self.case_sensitive else resolved_name
E           AttributeError: 'AliasChoices' object has no attribute 'lower'
```

</details>

### Example Code

```python
import pytest

from pydantic import AliasChoices, Field
# from pydantic import BaseModel
from pydantic_settings import BaseSettings


def test_settings():
    class Check(BaseSettings):
    # class Check(BaseModel):  # This works
        field: str = Field(
            "foo",
            alias=AliasChoices(
                "ALIAS_NAME"
            )
        )

    assert Check().field == 'foo'
    assert Check(ALIAS_NAME="bar").field == 'bar'
```

### Expect

For this test to pass with BaseSettings as it does so with BaseModel.

```shell
Platform: Ubuntu 22.04.4 LTS (Jammy Jellyfish)
Python version: Python 3.11.9 (main, Apr  6 2024, 17:59:24) [GCC 11.4.0]

pydantic           2.7.3
pydantic_core      2.18.4
pydantic-settings  2.3.0
```

--- a/pydantic_settings/main.py
+++ b/pydantic_settings/main.py
@@ -256,22 +256,6 @@ class BaseSettings(BaseModel):
 
         # Configure built-in sources
         init_settings = InitSettingsSource(self.__class__, init_kwargs=init_kwargs)
-        cli_settings = (
-            CliSettingsSource(
-                self.__class__,
-                cli_prog_name=cli_prog_name,
-                cli_parse_args=cli_parse_args,
-                cli_parse_none_str=cli_parse_none_str,
-                cli_hide_none_type=cli_hide_none_type,
-                cli_avoid_json=cli_avoid_json,
-                cli_enforce_required=cli_enforce_required,
-                cli_use_class_docs_for_groups=cli_use_class_docs_for_groups,
-                cli_prefix=cli_prefix,
-                case_sensitive=case_sensitive,
-            )
-            if cli_settings_source is None
-            else cli_settings_source
-        )
         env_settings = EnvSettingsSource(
             self.__class__,
             case_sensitive=case_sensitive,
@@ -305,7 +289,23 @@ class BaseSettings(BaseModel):
             file_secret_settings=file_secret_settings,
         )
         if not any([source for source in sources if isinstance(source, CliSettingsSource)]):
-            if cli_parse_args or cli_settings_source:
+            if cli_parse_args is not None or cli_settings_source is not None:
+                cli_settings = (
+                    CliSettingsSource(
+                        self.__class__,
+                        cli_prog_name=cli_prog_name,
+                        cli_parse_args=cli_parse_args,
+                        cli_parse_none_str=cli_parse_none_str,
+                        cli_hide_none_type=cli_hide_none_type,
+                        cli_avoid_json=cli_avoid_json,
+                        cli_enforce_required=cli_enforce_required,
+                        cli_use_class_docs_for_groups=cli_use_class_docs_for_groups,
+                        cli_prefix=cli_prefix,
+                        case_sensitive=case_sensitive,
+                    )
+                    if cli_settings_source is None
+                    else cli_settings_source
+                )
                 sources = (cli_settings,) + sources
         if sources:
             return deep_update(*reversed([source() for source in sources]))

